src/agora/docoding.cpp

```diff
@@ -49,35 +51,44 @@ EventData DoEncode::Launch(size_t tag) {
   size_t start_tsc = WorkerRdtsc();
 
   size_t symbol_idx_dl = cfg_->GetDLSymbolIdx(frame_id, symbol_id);
-  int8_t* input_ptr =
-      cfg_->GetInfoBits(raw_data_buffer_, symbol_idx_dl, ue_id, cur_cb_id);
-
-  LdpcEncodeHelper(ldpc_config.BaseGraph(), ldpc_config.ExpansionFactor(),
-                   ldpc_config.NumRows(), encoded_buffer_temp_, parity_buffer_,
-                   input_ptr);
-  int8_t* final_output_ptr = cfg_->GetEncodedBuf(
-      encoded_buffer_, frame_id, symbol_idx_dl, ue_id, cur_cb_id);
-  AdaptBitsForMod(reinterpret_cast<uint8_t*>(encoded_buffer_temp_),
-                  reinterpret_cast<uint8_t*>(final_output_ptr),
-                  BitsToBytes(ldpc_config.NumCbCodewLen()),
-                  cfg_->ModOrderBits());
-
-  // std::printf("Encoded data\n");
-  // int num_mod = LDPC_config.num_cb_codew_len() / cfg->mod_order_bits();
-  // for(int i = 0; i < num_mod; i++) {
-  //     std::printf("%u ", *(final_output_ptr + i));
-  // }
-  // std::printf("\n");
-
-  size_t duration = WorkerRdtsc() - start_tsc;
-  duration_stat_->task_duration_[0] += duration;
-  duration_stat_->task_count_++;
-  if (CyclesToUs(duration, cfg_->FreqGhz()) > 500) {
-    std::printf("Thread %d Encode takes %.2f\n", tid_,
-                CyclesToUs(duration, cfg_->FreqGhz()));
+  int8_t* input_ptr = new int8_t[cfg_->NumBytesPerCb()];
```

Comment:
We should avoid allocating memory here for performance reason. We can move it to the doEncode constructor and rename the buffer as something like `scrambled_buffer_`